mods/default/functions.lua

```diff
@@ -364,10 +356,24 @@ function default.register_fence(name, def)
 	-- Always add to the fence group, even if no group provided
 	def.groups.fence = 1
 
+	local material = def.material
 	def.texture = nil
 	def.material = nil
 
 	minetest.register_node(name, def)
+
+	-- Register crafting recipe, trim away starting colon if any
+	if not material then return end
```

Comment:
why would this be nil?

Reply:
Makes sense, just document it in `game_api.txt` too.

Reply:
Done in 3b39117

Weirdly only fence is listed in that API documentation, so only that one is fixed.